### PR TITLE
initial checkin to run ci-e2e

### PR DESCRIPTION
**Pull request by deepakm-ntnx**
**What this PR does / why we need it**:
This is required to integrate e2e run on every PR pre-submit job thru CI

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**How Has This Been Tested?**:
Internal Prow CI job was run with the script and found working.

_Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and test output_

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Review by tuxtof — DISMISSED** [review]
LGTM

File: Makefile [hunk 1]
```diff
@@ -318,7 +318,7 @@ test-kubectl-workload: ## Run kubectl queries to get all capx workload related o
 	kubectl --kubeconfig ./${TEST_CLUSTER_NAME}.workload.kubeconfig get nodes,ns
 
 .PHONY: test-e2e
-test-e2e: docker-build-e2e $(GINKGO) cluster-e2e-templates cluster-templates ## Run the end-to-end tests
```
Comment (adiantum) [inline Makefile:321]: There will be errors in Prow env with the modification.

Comment (yannickstruyf3) [inline Makefile:321]: Why is docker-build-e2e removed?

**Review by yannickstruyf3 — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (yannickstruyf3) [reply Makefile:321]: Do we still require docker-build-e2e if it is not used anymore?

**Review by yannickstruyf3 — DISMISSED** [review]

**Comment by adiantum** [comment]
LGTM

**Review by adiantum — APPROVED** [review]

**Review by tuxtof — APPROVED** [review]

**Review by yannickstruyf3 — APPROVED** [review]

**Pull request merged**